plugins/reporters/cyclonedx/src/main/kotlin/CycloneDxReporter.kt

```diff
@@ -369,7 +369,7 @@ class CycloneDxReporter : Reporter {
             val outputFile = outputDir.resolve("$outputName.$fileExtension")
 
             val bomGenerator = when (fileExtension) {
-                "xml" -> BomGeneratorFactory.createXml(schemaVersion, bom)
+                "xml" -> BomGeneratorFactory.createXml(schemaVersion, bom).toXmlString()
```

Comment:
Should the exception be caught, so that the failure remains limited to one file extension, but allows other file extensions to succeed?

Reply:
I've added a new commit to generally address that.